src/Examine/LuceneEngine/Providers/LuceneIndex.cs

```diff
@@ -1240,12 +1374,44 @@ private IndexWriter CreateIndexWriter()
         protected virtual IndexWriter WriterFactory(Directory d)
         {
             if (d == null) throw new ArgumentNullException(nameof(d));
-            var writer = new IndexWriter(d, FieldAnalyzer, false, IndexWriter.MaxFieldLength.UNLIMITED);
 
-            // clear out current scheduler and set the error logging one
-            writer.MergeScheduler.Dispose();
-            writer.SetMergeScheduler(new ErrorLoggingConcurrentMergeScheduler(Name,
-                (s, e) => OnIndexingError(new IndexingErrorEventArgs(this, s, "-1", e))));
+            IndexWriter writer;
+            if (_directory is ExamineDirectory examineDirectory)
+            {
+                if (examineDirectory.GetDeletionPolicy() != null)
+                {
+                    //This line is creating a lock on read directories
+                    //Check dirty first so the folder is not locked.
+                    examineDirectory.CheckDirtyWithoutWriter();
+                    writer = new ExamineIndexWriter(d, FieldAnalyzer, false, examineDirectory.GetDeletionPolicy(),
+                        IndexWriter.MaxFieldLength.UNLIMITED);
+                }
+                else
+                {
+                    writer = new ExamineIndexWriter(d, FieldAnalyzer, false, IndexWriter.MaxFieldLength.UNLIMITED);
+                }
+
+                if (examineDirectory.GetMergeScheduler() != null)
```

Comment:
Minor - but calling a method multiple times shouldn't be done since normally that will incur some performance. Better to store the result of GetMergeScheduler() and re-use it.